### ISSUE TITLE
Issue of loading data

### ISSUE THREAD

**Issue by weiren1998** [cioppaanthony/context-aware-loss#2]
Hi, I am trying to run the train.py on my own server, but I have some problems when I load the data named 'listgame_Train_300_ResNET_PCA512_labels.npy'. Can you help me to solve this problem? 
The code of train.py stpos running at this line  "self.labels = np.load(file_path_labels, allow_pickle=True)", and the error is showed below. (The version of numpy I used is 1.17.3.) 
![image](https://user-images.githubusercontent.com/45010350/84050659-97a28380-a9e0-11ea-9211-78067907fee7.png)
Looking forward to you reply!

**Comment by cioppaanthony** [comment]
Hi, so you went with option 2 for downloading the data (only ResNet_PCA512 data from the drive). 
In the code, The loading of the labels comes after the loading of the features, so I guess that you were able to load the features correctly at the previous line ? Could you please confirm that ?
I saved both arrays the same way using np.save(path, array). You can see it commented on lines 146-151 in utils/io_module.py
I will try to figure out where this issue comes from. Could you please share your complete conda environment (as we did in the readme) so that I can reproduce the error? A solution for you in the meantime would be to download the dataset from option 1. This skips the loading of the pre-registered arrays.

**Comment by weiren1998** [comment]
I am trying to use the option 2 for loading training dataset because the size of the dataset of option 1 is too large to download from the internet easily. What your guess is right. The process of loading features is correct but the labels is failed. I saw the comments on lines 146-151 in utils/io_module.py, but I still confused that I can't load the labels correctly.

**Comment by cioppaanthony** [comment]
That is very intriguing as these two arrays were saved the same way. I will try to reproduce the error and keep you updated. :-)

**Comment by weiren1998** [comment]
I am trying to use the option 2 for loading the training data because the the size of dataset of option 1 is too large to download easily.
Your guess is right. What I comfused is I can load the features correctly but I can't load the labels correctly.
The details of conda environment I used is showing below:(I just list the packages you mentioned in README.MD document)
numpy==1.17.3
tabulate==0.8.5
tqdm==4.23.1
h5py==2.10.0
matplotlib==3.0.3
opencv-python-headless==4.2.0.34
opencv-contrib-python-headless==4.2.0.34
tensorflow==2.0.0

Btw, could you tell me the data form of the features and labels in .npy files? 
Hope for your reply, thank you.

**Comment by SilvioGiancola** [comment]
Hi @weiren1998, Can you please check the file `ResNET_PCA512.zip` has been downloaded properly? It should weight ~9.8GB. The file is automatically downloaded from [Google Drive](https://drive.google.com/uc?id=1GeohUJww3U_F4oacsc3nRU_PJbf-Xqp-), can you check whether the service is available in your region? Were the `.npy` files properly extracted?

**Comment by weiren1998** [comment]
Hi @SilvioGiancola, Thank you for your reply. I have been checked the size of `ResNET_PCA512.zip` file. It is ~9.8GB. I am sure that I downloaded the data from the Google Drive and I think the `.npy` files extracted properly. 
![image](https://user-images.githubusercontent.com/45010350/84220353-24def880-ab05-11ea-817e-e64fd821a6e3.png)

**Comment by SilvioGiancola** [comment]
Hi @weiren1998 , I have not been able to reproduce your error. I have downloaded `ResNET_PCA512.zip` from GDrive, extracted the in the root folder of this GitHub repo and was able to train using `python src/train.py -d . -s weights`, that look for the `.npy` files on the root folder (`-d .`) and save the models in the folder `weights`, automatically created (`-s weights`).

Those are the file you need to have:
```
-rw-r--r--  1 giancos giancos 2.2G Apr  2 16:23 listgame_Test_100_ResNET_PCA512_features.npy
-rw-r--r--  1 giancos giancos  26M Apr  2 16:23 listgame_Test_100_ResNET_PCA512_labels.npy
-rw-r--r--  1 giancos giancos 6.3G Apr  2 16:19 listgame_Train_300_ResNET_PCA512_features.npy
-rw-r--r--  1 giancos giancos  76M Apr  2 16:19 listgame_Train_300_ResNET_PCA512_labels.npy
-rw-r--r--  1 giancos giancos 2.1G Apr  2 16:20 listgame_Valid_100_ResNET_PCA512_features.npy
-rw-r--r--  1 giancos giancos  26M Apr  2 16:20 listgame_Valid_100_ResNET_PCA512_labels.npy
```

Note that you don't necessarily need a GPU, so you can run the following: `CUDA_DEVICE_ORDER=PCI_BUS_ID CUDA_VISIBLE_DEVICES=-1 python src/train.py -d . -s weights`.

Also, make sure you are in your conda environment:
```
conda create -n context_aware
conda activate context_aware
conda install python=3.7 pip cudnn cudatoolkit=10.1
pip install numpy==1.17.3 tabulate==0.8.5 tqdm==4.23.1 h5py==2.10.0 matplotlib==3.0.3 opencv-python-headless==4.1.2.30 opencv-contrib-python-headless==4.1.2.30 tensorflow==2.1.0 scikit-video
```

Best of luck.

**Comment by weiren1998** [comment]
Hi @SilvioGiancola . Unfortunately, I followed the methods you told me and prepared the conda environment(using the same packages version as yours) again, but there was still the same error as before. I tried to use following code to read and print each labels and features files downloaded from GDrive into a `.txt` file.

> labels = np.load("listgame_***_*00_ResNET_PCA512_##.npy", allow_pickle=True)
> doc = open('***_##.txt', 'a')  
> print(labels, file=doc)

There are two files load unsuccessful, can you upload this two files(unzipped file) again to the GDrive. They are `listgame_Train_300_ResNET_PCA512_labels.npy` and `listgame_Valid_100_ResNET_PCA512_labels.npy`
Thank you very much!!!

Supplement:
I have been checked the size of the `.npy` files which is the same as yours.
> -rw-rw-r-- 1 renwei renwei 2.2G Jun  5 12:58 listgame_Test_100_ResNET_PCA512_features.npy
> -rw-rw-r-- 1 renwei renwei 26M Jun  5 12:58 listgame_Test_100_ResNET_PCA512_labels.npy
> -rw-rw-r-- 1 renwei renwei 6.3G Jun  5 13:21 listgame_Train_300_ResNET_PCA512_features.npy
> -rw-rw-r-- 1 renwei renwei 76M Jun  8 23:11 listgame_Train_300_ResNET_PCA512_labels.npy
> -rw-rw-r-- 1 renwei renwei 2.1G Jun  5 12:48 listgame_Valid_100_ResNET_PCA512_features.npy
> -rw-rw-r-- 1 renwei renwei 26M Jun  5 12:48 listgame_Valid_100_ResNET_PCA512_labels.npy

**Comment by weiren1998** [comment]
@SilvioGiancola @cioppaanthony Thank you for your patience. I download the data from GDrive again. And this problem is solved immediately. I don't know why but it worked. Maybe there are something loss during the data transmission.
I am now trying to follow your job and trying to reproduce your results. Thank you again.

**Comment by SilvioGiancola** [comment]
I'm glad it's solved. Issue closed :)
